### PR TITLE
Separate chart version replacer

### PR DESCRIPTION
**Pull request by xin-hedera**
**Detailed description**:

- Use a separate execution to replace chart version
- Use start of line anchor `^` to match `appVersion` and `version`

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:
Some dependency chart versions are now unquoted, so the replacement regex for the maven replacer plugin now accidentally pick up those and update them to `release.version`.

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: charts/hedera-mirror-common/Chart.yaml [hunk 1]
```diff
@@ -3,25 +3,25 @@ appVersion: "0.32.0-rc1"
 dependencies:
   - name: loki
     condition: loki.enabled
-    version: 2.5.0
+    version: "2.5.0"
```
Comment (Nana-EC) [inline charts/hedera-mirror-common/Chart.yaml:6]: I see the workaround but then it makes me wonder why in our regular replaces these versions aren't affected?
How does you other PR modify things to make these eligible for replacement?
If we get that down then maybe this workaround isn't needed

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Pondering if this workaround is best.
Might not need this change

File: charts/hedera-mirror-common/Chart.yaml (hunk 1 above)
Comment (Nana-EC) [reply charts/hedera-mirror-common/Chart.yaml:6]: I see the context noted in the other PR now. We regressed by removing the quotations that we had around the range.

I believe Steven said he removed the version ranges to ensure `helm dep build` worked in CI/CD.
I'm presuming the quotation marks don't break that flow and it was just the version range.
In that case this is good.
Just verify that

**Review by Nana-EC — DISMISSED** [review]
LGTM
Just verify that the quotes don't break the CI/CD flow noted in the PR that removed the quotes initially.

**Review by ijungmann — DISMISSED** [review]

File: charts/hedera-mirror-common/Chart.yaml [hunk 2]
```diff
@@ -3,25 +3,25 @@ appVersion: "0.32.0-rc1"
...
     repository: https://grafana.github.io/helm-charts
   - condition: prometheus-adapter.enabled
     name: prometheus-adapter
     repository: https://prometheus-community.github.io/helm-charts
-    version: 2.12.1
+    version: "2.12.1"
```
Comment (steven-sheehy) [inline charts/hedera-mirror-common/Chart.yaml:11]: This is a bit of a hack that's will probably break in the future when someone forgets that quotes means don't replace version. We should tweak the replacer to look only for `version` and `appVersion` at the beginning of the line (e.g. no spaces) so it doesn't replace for dependencies. This should solve the problem more future proof fashion.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: charts/hedera-mirror-common/Chart.yaml (hunk 2 above)
Comment (Nana-EC) [reply charts/hedera-mirror-common/Chart.yaml:11]: I pondered this also but I'm not sure it also scales.
See the recent bump - https://github.com/hashgraph/hedera-mirror-node/pull/1867/files
Open api and rosetta files all have version not at the beginning of the line. So they may be missed in the replacer.

Agreed regex is a preferred way as it's central. Hopefully there's a regex that works that isn't also brittle

Comment (xin-hedera) [reply charts/hedera-mirror-common/Chart.yaml]: there is another surprise:
```xml
<replacement>
    <token><![CDATA[(?<=\sversion: )[0-9a-zA-Z.-]+]]></token>
    <value>${release.chartVersion}</value>
</replacement>
<replacement>
    <token><![CDATA[(?<=(\s{3})version: )[0-9a-zA-Z.-]+]]></token>
    <value>${release.version}</value>
</replacement>
```

The two both update the unquoted dependency versions in `Chart.yaml`.

I end up using a separate execution for charts.yaml and restrict the regex to match start of line.

File: pom.xml [hunk 3]
```diff
@@ -444,47 +444,67 @@
...
+                                    <goal>replace</goal>
+                                </goals>
+                                <inherited>false</inherited>
+                                <phase>prepare-package</phase>
+                                <configuration>
+                                    <includes>
+                                        <include>charts/**/Chart.yaml</include>
+                                    </includes>
+                                    <regexFlags>
+                                        <regexFlag>MULTILINE</regexFlag>
```
Comment (steven-sheehy) [inline pom.xml:461]: It doesn't match across lines, does it?

Comment (steven-sheehy) [reply pom.xml:461]: Ah, true. Forgot about that.

**Review by steven-sheehy — APPROVED** [review]

**Review by ijungmann — DISMISSED** [review]

**Review by ijungmann — APPROVED** [review]

**Review by Nana-EC — APPROVED** [review]

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**